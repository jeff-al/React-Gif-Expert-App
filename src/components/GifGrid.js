import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem"
import PropTypes from "prop-types";

const GifGrid = ( { category } ) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h2>{ category }</h2>  

            { loading && <p>Loading...</p>}

            {<div className="card-grid">
                { 
                    images.map( ( img ) => <GifGridItem 
                        key={ img.id } 
                        { ...img }
                        /> 
                    )
                }
            </div>}
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;
