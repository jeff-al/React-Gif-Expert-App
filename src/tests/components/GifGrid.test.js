import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('GifGrid Tests', () => {
    
    const category = "Keylor";

    test('should match snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            value: true
        })

        const wrapper = shallow( <GifGrid category={ category }/>);
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show items after load', () => {
        
        const gifs = [{
            id: "AAA",
            url: "https://test.jpg",
            title: "Test"
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            value: false
        })

        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("p").exists() ).toBe( false );
        expect( wrapper.find("GifGridItem").length ).toBe( gifs.length )

    });
    

})
