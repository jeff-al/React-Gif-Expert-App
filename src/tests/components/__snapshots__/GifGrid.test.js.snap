// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Tests should match snapshot 1`] = `
<Fragment>
  <h2>
    Keylor
  </h2>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Tests should show items after load 1`] = `
<Fragment>
  <h2>
    Keylor
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="AAA"
      key="AAA"
      title="Test"
      url="https://test.jpg"
    />
  </div>
</Fragment>
`;
